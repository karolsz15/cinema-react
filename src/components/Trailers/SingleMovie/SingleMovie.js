import React, {Component} from 'react';
import MovieModal from '../../UI/MovieModal/MovieModal';

class SingleMovie extends Component {

    state = {
        modalVisible: false
    }

    showModal =() => {
        this.setState({modalVisible: true})
    };

    hideModal = () => {
        this.setState({modalVisible: false})
    }

    render() {
        return (
            <React.Fragment>

                <MovieModal 
                    show={this.state.modalVisible} 
                    onHide={this.hideModal} 
                    title={this.props.title} 
                    trailer={this.props.trailer} 
                />

                <div className="content-grid">
                    <div className="play-icon"><img src={this.props.poster} alt="movie poster" /></div>
                    <h3>{this.props.summary}</h3>
                    <a onClick={this.showModal} className="button play-icon" href="#small-dialog3">Watch trailer</a>
                    <a className="button play-icon" href="/booking">Book now</a>
                </div>    
            </React.Fragment>
        );
    }
};

export default SingleMovie;