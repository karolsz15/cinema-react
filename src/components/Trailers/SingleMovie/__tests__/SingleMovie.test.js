import React from 'react';
import { shallow } from 'enzyme';
import SingleMovie from '../SingleMovie';

let wrapper = shallow(<SingleMovie />);

describe('SingleMovie component', () => {
  it('should render the SingleMovie correctly', () => {   
    expect(wrapper).toMatchSnapshot();
  });
});