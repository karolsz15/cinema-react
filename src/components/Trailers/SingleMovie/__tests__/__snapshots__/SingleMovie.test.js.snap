// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleMovie component should render the SingleMovie correctly 1`] = `
<Fragment>
  <MovieModal
    onHide={[Function]}
    show={false}
  />
  <div
    className="content-grid"
  >
    <div
      className="play-icon"
    >
      <img
        alt="movie poster"
        className="trailerPoster"
      />
    </div>
    <div
      className="trailerDescription"
    />
    <a
      className="button play-icon"
      href="#trailer"
      onClick={[Function]}
    >
      Watch trailer
    </a>
    <a
      className="button play-icon"
      href="/booking"
    >
      Book now
    </a>
  </div>
</Fragment>
`;
